examples/vector_databases/Vector_db_introduction.ipynb

```diff
@@ -0,0 +1,1031 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 101,
+   "id": "5fc1b083",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Uploading vectors to title namespace..\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Upsert title vectors in title namespace\n",
+    "# NOTE: Using a thread pool here can accelerate this upsert operation\n",
+    "print(\"Uploading vectors to title namespace..\")\n",
+    "for batch_df in df_batcher(article_df):\n",
+    "    index.upsert(vectors=zip(batch_df.vector_id, batch_df.title_vector), namespace='title')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 102,
+   "id": "f90c7fba",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'dimension': 1536,\n",
+       " 'index_fullness': 0.2,\n",
+       " 'namespaces': {'content': {'vector_count': 50000},\n",
+       "                'title': {'vector_count': 50000}},\n",
+       " 'total_vector_count': 100000}"
+      ]
+     },
+     "execution_count": 102,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Check index size for each namespace\n",
+    "index.describe_index_stats()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2da40a69",
+   "metadata": {},
+   "source": [
+    "### Search data\n",
+    "\n",
+    "Now we'll enter some dummy searches and check we get decent results back"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 103,
+   "id": "d701b3c7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# First we'll create dictionaries mapping vector IDs to their outputs so we can retrieve the text for our search results\n",
+    "titles_mapped = dict(zip(article_df.vector_id,article_df.title))\n",
+    "content_mapped = dict(zip(article_df.vector_id,article_df.text))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 104,
+   "id": "3c8c2aa1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def query_article(query, namespace, top_k=5):\n",
+    "    '''Queries an article using its title in the specified\n",
+    "     namespace and prints results.'''\n",
+    "\n",
+    "    # Create vector embeddings based on the title column\n",
+    "    embedded_query = openai.Embedding.create(\n",
+    "                                            input=query,\n",
+    "                                            model=MODEL,\n",
+    "                                            )[\"data\"][0]['embedding']\n",
+    "\n",
+    "    # Query namespace passed as parameter using title vector\n",
+    "    query_result = index.query(embedded_query, \n",
+    "                               namespace=namespace, \n",
+    "                               top_k=top_k)\n",
+    "\n",
+    "    # Print query results \n",
+    "    print(f'\\nMost similar results querying {query} in \"{namespace}\" namespace:\\n')\n",
```

Comment:
would wrap the {query} in quotation marks too

Reply:
maybe rephrase to 'Most similar results to "{query}" in "{namespace}" namespace:\n'. Current phrasing feels a bit harder to parse with the querying verb thrown in.

Reply:
Updated